src/lib/blkalloc/append_blk_allocator.cpp

```diff
@@ -102,86 +80,68 @@ BlkAllocStatus AppendBlkAllocator::alloc(blk_count_t nblks, const blk_alloc_hint
     }
 
     // Push 1 blk to the vector which has all the requested nblks;
-    out_bid = BlkId{m_last_append_offset, nblks, m_chunk_id};
-
-    auto cur_cp = hs()->cp_mgr().cp_guard();
-    m_last_append_offset += nblks;
-    m_freeable_nblks -= nblks;
-
-    // it is guaranteed that dirty buffer always contains updates of current_cp or next_cp, it will
-    // never get dirty buffer from across updates;
-    set_dirty_offset(cur_cp->id() % MAX_CP_COUNT);
+    out_bid = BlkId{m_last_append_offset.fetch_add(nblks), nblks, m_chunk_id};
```

Comment:
should we mark `m_is_dirty` to true here?

Reply:
It is not dirtied yet, dirty is relevant only when we commit the blk. In other words, dirty phase is between committing the blk and persisting the blk in disk. So alloc_on_disk() has to be done dirtying it.

Reply:
I understand your point. but I thinks that is only right to raft_repl_dev. for raft_repl_dev, when applying log in background thread, `on_commit` will be called and thus `data_service().commit_blk` and `alloc_on_disk()`, this is  the scenario you mentioned here.

but , data_service can be used independently , if we use data_service with append_block_allocator(not in raft_repl_dev like nublox), then we will only call `async_allocate_write` , and `commit_blk` will not be called in this case, right? so the append_block_allocator will not be marked as dirty and cp flush will do nothing , which will lead to data loss.

Reply:
The data service expects one to use journal service or maintain their own journal. There is no way for data service can keep track of pending commits. If one were to use Data service alone without other service, then it can be done with no crash protection.

Reply:
ok , if this is the design goal and the contract between homestore and the related user, it makes sense. got it ! thanks